### PR TITLE
SCT-143 Build: Template file preprocessor to simplify template file development

### PR DESCRIPTION
**Pull request by gaoweiqiao-tre**
## Related Task

- [Build: Template file preprocessor to simplify template file development](https://retailai.atlassian.net/browse/SCT-143)

## Details

- remove file subfix `.tpl`
- define `/**#bean*/  /*#bean.replace()**/` preprocess macro
- develop preprocessor

### CODE REVIEW

File: internal/project/framework/bootstrap/kernel.go [hunk 1]
```diff
@@ -9,14 +9,15 @@ import (
...
-	"{{ .PkgPath }}/framework/dbdrivers"
-	"{{ .PkgPath }}/framework/internals/binder"
-	ierror "{{ .PkgPath }}/framework/internals/error"
-	"{{ .PkgPath }}/framework/internals/global"
-	"{{ .PkgPath }}/framework/internals/helpers"
-	imiddleware "{{ .PkgPath }}/framework/internals/middleware"
-	str "{{ .PkgPath }}/framework/internals/string"
-	"{{ .PkgPath }}/framework/internals/template"
+	/**#bean*/
+	"demo/framework/dbdrivers" /*#bean.replace("{{ .PkgPath }}/framework/dbdrivers")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/bootstrap/kernel.go:13]: we should put these two lines in one line, right?

File: internal/project/framework/dbdrivers/mongo.go [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 package dbdrivers
 
 import (
 	"context"
 	"encoding/json"
 	"time"
 
-	"{{ .PkgPath }}/framework/internals/aes"
+	/**#bean*/
+	"demo/framework/internals/aes" /*#bean.replace("{{ .PkgPath }}/framework/internals/aes")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/dbdrivers/mongo.go:10]: put these two lines in one line

File: internal/project/framework/dbdrivers/mysql.go [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
+/**#bean*/ /*#bean.replace({{ .Copyright }})**/
 package dbdrivers
 
 import (
 	"encoding/json"
 	"fmt"
 
-	"{{ .PkgPath }}/framework/internals/aes"
+	/**#bean*/
+	"demo/framework/internals/aes" /*#bean.replace("{{ .PkgPath }}/framework/internals/aes")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/dbdrivers/mysql.go:9]: put these two lines in one line

File: internal/project/framework/dbdrivers/redis.go [hunk 4]
```diff
@@ -1,10 +1,11 @@
-{{ .Copyright }}
+/**#bean*/ /*#bean.replace({{ .Copyright }})**/
 package dbdrivers
 
 import (
 	"encoding/json"
 
-	"{{ .PkgPath }}/framework/internals/aes"
+	/**#bean*/
+	"demo/framework/internals/aes" /*#bean.replace("{{ .PkgPath }}/framework/internals/aes")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/dbdrivers/redis.go:8]: put these two lines in one line

File: internal/project/framework/internals/async/async.go [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
 // Safe way to execute `go routine` without crashing the parent process while having a `panic`.
 package async
 
 import (
 	"fmt"
 	"reflect"
 
-	"{{ .PkgPath }}/framework/internals/global"
+	/**#bean*/
+	"demo/framework/internals/global" /*#bean.replace("{{ .PkgPath }}/framework/internals/global")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/internals/async/async.go:10]: put these two lines in one line

File: internal/project/framework/internals/binder/bind.go [hunk 6]
```diff
@@ -9,8 +9,9 @@ import (
 	"net/http"
 	"strings"
 
-	"{{ .PkgPath }}/framework/internals/helpers"
-	structure "{{ .PkgPath }}/framework/internals/struct"
+	/**#bean*/
+	"demo/framework/internals/helpers" /*#bean.replace("{{ .PkgPath }}/framework/internals/helpers")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/internals/binder/bind.go:13]: put these two lines in one line

File: internal/project/framework/internals/cli/cli.go [hunk 7]
```diff
@@ -1,13 +1,14 @@
...
 
 import (
 	"fmt"
 	"os"
 
-	"{{ .PkgPath }}/framework/internals/chalk"
-	"{{ .PkgPath }}/framework/internals/global"
-	"{{ .PkgPath }}/framework/internals/sentry"
+	/**#bean*/
+	"demo/framework/internals/chalk" /*#bean.replace("{{ .PkgPath }}/framework/internals/chalk")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/internals/cli/cli.go:9]: put these two lines in one line

File: internal/project/framework/internals/error/error.go [hunk 8]
```diff
@@ -7,7 +7,8 @@ import (
 	"fmt"
 	"io"
 
-	"{{ .PkgPath }}/framework/internals/stacktrace"
+	/**#bean*/
+	"demo/framework/internals/stacktrace" /*#bean.replace("{{ .PkgPath }}/framework/internals/stacktrace")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/internals/error/error.go:11]: put these two lines in one line

File: internal/project/framework/internals/error/error_handler.go [hunk 9]
```diff
@@ -1,12 +1,13 @@
...
 package error
 
 import (
 	"net/http"
 	"strings"
 
-	"{{ .PkgPath }}/framework/internals/sentry"
-	"{{ .PkgPath }}/framework/internals/validator"
+	/**#bean*/
+	"demo/framework/internals/sentry" /*#bean.replace("{{ .PkgPath }}/framework/internals/sentry")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/internals/error/error_handler.go:9]: put these two lines in one line

File: internal/project/framework/internals/helpers/common.go [hunk 10]
```diff
@@ -8,9 +8,10 @@ import (
 	"os"
 	"time"
 
-	"{{ .PkgPath }}/framework/dbdrivers"
-	"{{ .PkgPath }}/framework/internals/global"
-	str "{{ .PkgPath }}/framework/internals/string"
+	/**#bean*/
+	"demo/framework/dbdrivers" /*#bean.replace("{{ .PkgPath }}/framework/dbdrivers")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/internals/helpers/common.go:12]: put these two lines in one line

File: internal/project/framework/internals/latency/latency.go [hunk 11]
```diff
@@ -1,12 +1,13 @@
...
 package latency
 
 import (
 	"bytes"
 	"encoding/binary"
 	"time"
 
-	"{{ .PkgPath }}/framework/internals/global"
+	/**#bean*/
+	"demo/framework/internals/global" /*#bean.replace("{{ .PkgPath }}/framework/internals/global")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/internals/latency/latency.go:10]: put these two lines in one line

File: internal/project/framework/internals/middleware/latency.go [hunk 12]
```diff
@@ -1,13 +1,14 @@
...
 
 import (
 	"fmt"
 	"net/http"
 	"time"
 
-	"{{ .PkgPath }}/framework/internals/async"
-	"{{ .PkgPath }}/framework/internals/latency"
+	/**#bean*/
+	"demo/framework/internals/async" /*#bean.replace("{{ .PkgPath }}/framework/internals/async")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/internals/middleware/latency.go:10]: put these two lines in one line

File: internal/project/framework/internals/middleware/method_not_allowed_route_not_found.go [hunk 13]
```diff
@@ -1,12 +1,13 @@
...
 package middleware
 
 import (
 	"net/http"
 	"sort"
 	"strings"
 
-	ierror "{{ .PkgPath }}/framework/internals/error"
+	/**#bean*/
+	ierror "demo/framework/internals/error" /*#bean.replace(ierror "{{ .PkgPath }}/framework/internals/error")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/internals/middleware/method_not_allowed_route_not_found.go:11]: put these two lines in one line

File: internal/project/framework/internals/middleware/server_stats.go [hunk 14]
```diff
@@ -8,8 +8,9 @@ import (
 	"sync"
 	"time"
 
-	ierror "{{ .PkgPath }}/framework/internals/error"
-	"{{ .PkgPath }}/framework/internals/latency"
+	/**#bean*/
+	ierror "demo/framework/internals/error" /*#bean.replace(ierror "{{ .PkgPath }}/framework/internals/error")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/internals/middleware/server_stats.go:12]: put these two lines in one line

File: internal/project/framework/internals/sentry/sentry.go [hunk 15]
```diff
@@ -1,11 +1,12 @@
...
+/**#bean*/ /*#bean.replace({{ .Copyright }})**/
 package sentry
 
 import (
 	"errors"
 	"reflect"
 
-	"{{ .PkgPath }}/framework/internals/global"
+	/**#bean*/
+	"demo/framework/internals/global" /*#bean.replace("{{ .PkgPath }}/framework/internals/global")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/framework/internals/sentry/sentry.go:9]: put these two lines in one line

File: internal/project/handlers/mytest.go [hunk 16]
```diff
@@ -1,10 +1,11 @@
-{{ .Copyright }}
+/**#bean*/ /*#bean.replace({{ .Copyright }})**/
 package handlers
 
 import (
 	"net/http"
 
-	"{{ .PkgPath }}/services"
+	/**#bean*/
+	"demo/services" /*#bean.replace("{{ .PkgPath }}/services")**/
```
Comment (SunJinping-TRE-C) [inline internal/project/handlers/mytest.go:8]: put these two lines in one line

File: internal/project/framework/bootstrap/kernel.go (hunk 1 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/bootstrap/kernel.go:13]: `go fmt` command make it, `go fmt` will run before I commit

File: internal/project/framework/dbdrivers/mongo.go (hunk 2 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/dbdrivers/mongo.go:10]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/framework/dbdrivers/mysql.go (hunk 3 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/dbdrivers/mysql.go:9]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/framework/dbdrivers/redis.go (hunk 4 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/dbdrivers/redis.go:8]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/framework/internals/async/async.go (hunk 5 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/internals/async/async.go:10]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/framework/internals/binder/bind.go (hunk 6 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/internals/binder/bind.go:13]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/framework/internals/cli/cli.go (hunk 7 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/internals/cli/cli.go:9]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/framework/internals/error/error.go (hunk 8 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/internals/error/error.go:11]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/framework/internals/error/error_handler.go (hunk 9 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/internals/error/error_handler.go:9]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/framework/internals/helpers/common.go (hunk 10 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/internals/helpers/common.go:12]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/framework/internals/latency/latency.go (hunk 11 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/internals/latency/latency.go:10]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/framework/internals/middleware/latency.go (hunk 12 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/internals/middleware/latency.go:10]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/framework/internals/middleware/method_not_allowed_route_not_found.go (hunk 13 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/internals/middleware/method_not_allowed_route_not_found.go:11]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/framework/internals/middleware/server_stats.go (hunk 14 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/internals/middleware/server_stats.go:12]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/framework/internals/sentry/sentry.go (hunk 15 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/internals/sentry/sentry.go:9]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/handlers/mytest.go (hunk 16 above)
Comment (gaoweiqiao-tre) [reply internal/project/handlers/mytest.go:8]: `go fmt` command make it, `go fmt` will run before I commit,but I don't know the fmt rules
bean macro still works in two lines

File: internal/project/framework/bootstrap/kernel.go (hunk 1 above)
Comment (SunJinping-TRE-C) [reply internal/project/framework/bootstrap/kernel.go:13]: yes, it works but not very beautiful... it's better to cancel `go fmt` before commit and push again.
@tanvir-retailai @yu-yk what do you guys think?

**Review by yu-yk — APPROVED** [review]

**Pull request merged**